subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultIvyArtifactRepository.java

```diff
@@ -300,6 +296,7 @@ public void layout(String layoutName) {
                 layout = instantiator.newInstance(MavenRepositoryLayout.class);
                 break;
             case "pattern":
+                // This case should have been deprecated but was not
```

Comment:
The problem is that I found usage of `layout "pattern"` in our code. Simply removing is an issue because it changes the behavior silently.
Another alternative would be to throw in case `pattern` is used as a value, assuming not many users did that without also configuring patterns.

Reply:
Decided to throw